Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.cpp

```diff
@@ -198,32 +198,44 @@ namespace AzToolsFramework
 
         AZStd::unique_ptr<AZ::Entity> Instance::DetachEntity(const AZ::EntityId& entityId)
```

Comment:
question: Is there an existing test or a test we could add to verify this function to ensure it is doing what's expected? It might be cool to have a test that verified the previous error state and then run it again with this change to show it passes? It might be tricky to do but wanted to check as it could be really helpful in future

Reply:
I'll look into adding a test.

Reply:
Do we need this change at all? So far, DetachEntity() only detaches entities within a prefab except the container entity. And if it is a container entity, users can call DetachContainerEntity() themselves. Any reason for changing this behavior now.
It is similar to other calls in the api. GetEntities() only returns entities in the prefab instance except the container entity.

Reply:
I think that's fine for now because that's how the API calls are defined so far: entities and container entity have different functions. We can change that behavior in the future if needed and let that be consistent across all functions Detach, Get, etc. but that'll be a breaking change since existing functionality might have made different calls for them separately and we need to take that into account when making the change. It'll also need to be accompanied with a 'impactful change' announcement

Reply:
OK, I left the DetachEntity function as is and moved the new logic outside of it.